### PR TITLE
TRANSIP: Fix TXT quoting

### PR DESCRIPTION
**Pull request by blackshadev**
Fixes !2702

- Fix TXT quoting, by not quoting at all. TransIP doesn't need it (any more)
- Fix and added some auditrecord entries to let integration tests pass
  - A TransIP TXT record cannot be empty
  - TransIP trims spaces from TXT records.

### CODE REVIEW

**Review by cafferata — APPROVED** [review]

**Comment by cafferata** [comment]
```shell
git remote add blackshadev git@github.com:blackshadev/dnscontrol.git
git fetch blackshadev
git checkout -b fix-quoting --track blackshadev/fix-quoting 
go build
```
```shell
dnscontrol preview --domains jcid.nl
```
```text
******************** Domain: jcid.nl
Done. 0 corrections.
```

**Comment by blackshadev** [comment]
> ```shell
> git remote add blackshadev git@github.com:blackshadev/dnscontrol.git
> git fetch blackshadev
> git checkout -b fix-quoting --track blackshadev/fix-quoting 
> go build
> ```
> 
> ```shell
> dnscontrol preview --domains jcid.nl
> ```
> 
> ```
> ******************** Domain: jcid.nl
> Done. 0 corrections.
> ```

So is this a good thing? 

Did `jcid.nl` contain a fixed record and is the 0 corrections as expected. Or did you expect it to remove some quoted TXT record?

**Comment by cafferata** [comment]
Your comment and my edit crossed paths. Fix looks good to me. I had already given a GitHub approval in advance. 👍

File: integrationTest/integration_test.go [hunk 1]
```diff
@@ -1126,6 +1126,7 @@ func makeTests(t *testing.T) []*TestGroup {
 			tc("TXT with semicolon ws", txt("foosc2", `wssemi ; colon`)),
 
 			tc("TXT interior ws", txt("foosp", "with spaces")),
+			// Some providers trim spaces
```
Comment (tlimoncelli) [inline integrationTest/integration_test.go:1129]:
```suggestion

						tc("TXT leading ws", txt("foowsb", " leadingspace")),
```

File: integrationTest/integration_test.go [hunk 2]
```diff
@@ -1126,6 +1126,8 @@ func makeTests(t *testing.T) []*TestGroup {
 			tc("TXT with semicolon ws", txt("foosc2", `wssemi ; colon`)),
 
 			tc("TXT interior ws", txt("foosp", "with spaces")),
+
+						tc("TXT leading ws", txt("foowsb", " leadingspace")),
```
Comment (tlimoncelli) [inline integrationTest/integration_test.go:1130]:
```suggestion

				tc("TXT leading ws", txt("foowsb", " leadingspace")),
```

File: integrationTest/integration_test.go [hunk 3]
```diff
@@ -1126,6 +1126,9 @@ func makeTests(t *testing.T) []*TestGroup {
 			tc("TXT with semicolon ws", txt("foosc2", `wssemi ; colon`)),
 
 			tc("TXT interior ws", txt("foosp", "with spaces")),
+
+
+				tc("TXT leading ws", txt("foowsb", " leadingspace")),
```
Comment (tlimoncelli) [inline integrationTest/integration_test.go:1131]:
```suggestion
			tc("TXT leading ws", txt("foowsb", " leadingspace")),

```

File: integrationTest/integration_test.go [hunk 4]
```diff
@@ -1126,6 +1126,9 @@ func makeTests(t *testing.T) []*TestGroup {
 			tc("TXT with semicolon ws", txt("foosc2", `wssemi ; colon`)),
 
 			tc("TXT interior ws", txt("foosp", "with spaces")),
+
+			tc("TXT leading ws", txt("foowsb", " leadingspace")),
+
 			tc("TXT trailing ws", txt("foows1", "with space at end ")),
```
Comment (tlimoncelli) [inline integrationTest/integration_test.go:1132]:
```suggestion
			tc("TXT interior ws", txt("foosp", "with spaces")),
			tc("TXT leading ws", txt("foowsb", " leadingspace")),
			tc("TXT trailing ws", txt("foows1", "trailingws ")),
```

**Pull request merged**